src/main/java/frc/robot/utils/MovingAverage.java

```diff
@@ -0,0 +1,114 @@
+package frc.robot.utils;
+
+import java.io.*;
+import java.util.Map;
+import java.util.TreeMap;
+
+/**
+ * The moving average contains a list of velocity based on the distance, and should give an approximation of the velocity to apply.
+ * In order to the moving average object to approximate better, you need to pass more values.
+ * The object calculate calculate a straight line between the closest point, and retrieve the velocity (in the straight line between the records).
+ */
+public class MovingAverage {
```

Comment:
That's a [linear regression](https://en.wikipedia.org/wiki/Linear_regression), not a moving average.
Your solution here is good, but usually, you find a line that the [sum of squared distances from it to each point is minimal](https://en.wikipedia.org/wiki/Least_squares). You can find such implementations online.
Moreover, I'm not sure a linear regression might be enough and you might need a [_polynomial_ regression](https://en.wikipedia.org/wiki/Polynomial_regression) of a higher degree. I suggest experimenting with the values in Excel to see what is the relationship between the distance and the desired velocity.

Reply:
I don't know what does 👀 mean. This is still not a moving average.